plasmapy/diagnostics/tests/test_thomson.py

```diff
@@ -215,6 +215,30 @@ def test_single_species_collective_spectrum(single_species_collective_spectrum):
         "instead of expected 17.7899"
     )
 
+def test_notched_spectrum(single_species_collective_spectrum):
```

Comment:
Good test! Since your code supports multiple notches, I think you should parametrize this test (using pytest parameterize) for two examples: 
1) what you have here - just one notch
2) A set of 2-3 notches 

That will verify the code is interpreting the specified notch input correctly